transmogrifier/sources/xml/marc.py

```diff
@@ -934,6 +889,61 @@ def get_literary_form(cls, source_record: Tag) -> str | None:
             return "Fiction"
         return None
 
+    @classmethod
+    def get_locations(cls, source_record: Tag) -> list[timdex.Location] | None:
+        locations = []
+        location_marc_fields = [
+            {
+                "tag": "751",
+                "subfields": "a",
+                "kind": "Geographic Name",
+            },
+            {
+                "tag": "752",
+                "subfields": "abcdefgh",
+                "kind": "Hierarchical Place Name",
+            },
+        ]
+
+        # get locations (place of publication) from control field 008/15-17
+        if (
+            fixed_location_code := cls._get_control_field(source_record)[15:18].strip()
+        ) and (
+            location_name := Marc.loc_crosswalk_code_to_name(
+                code=fixed_location_code,
+                crosswalk=cls.country_code_crosswalk,
+                record_id=cls.get_source_record_id(source_record),
+                code_type="country",
+            )
+        ):
+            locations.append(
+                timdex.Location(value=location_name, kind="Place of Publication")
+            )
+
+        # get locations from data fields
+        for location_marc_field in location_marc_fields:
+            locations.extend(
+                [
+                    timdex.Location(
+                        value=location_value.rstrip(" .,/)"),
+                        kind=location_marc_field["kind"],
+                    )
+                    for datafield in source_record.find_all(
+                        "datafield", tag=location_marc_field["tag"]
+                    )
+                    if (
+                        location_value := (
+                            cls.create_subfield_value_string_from_datafield(
+                                datafield,
+                                location_marc_field["subfields"],
+                                " - ",
+                            )
+                        )
+                    )
+                ]
+            )
```

Comment:
Hmm, I decided to [write a submethod for retrieving the "place of publication" from the control field](https://github.com/MITLibraries/transmogrifier/pull/202/files#diff-58573f6e57860b560075cbe825c45aace3f2830bb51d47f8d8dbf4b1b0b5d5ebR857-R870). 🤔 Since most of the field methods use the `create list -> for loop -> list.extend _if_`, I'd like to keep that in the main `get_locations` field method. Let me know what you think!

Reply:
That's fine, I appreciate the logical consistency!